cloudify_agent/installer/config/agent_config.py

```diff
@@ -150,15 +150,19 @@ def _set_name(self):
         self['name'] = name
 
     def _set_network(self):
-        networks = self.pop('networks')  # retrieved from the provider context
         network = self.setdefault('network', constants.DEFAULT_NETWORK_NAME)
-        manager_ip = networks.get(network)
-        if not manager_ip:
-            raise exceptions.AgentInstallerConfigurationError(
-                'The network associated with the agent (`{0}`) does not '
-                'appear in the list of manager networks assigned at bootstrap '
-                '({1})'.format(network, networks.keys())
-            )
+        networks = self.pop('networks', None)
+        if networks:
+            manager_ip = networks.get(network)
+            if not manager_ip:
+                raise exceptions.AgentInstallerConfigurationError(
+                    'The network associated with the agent (`{0}`) does not '
+                    'appear in the list of manager networks assigned at '
+                    'bootstrap ({1})'.format(network, ', '.join(networks))
+                )
+        else:
+            # Might be getting here when working in local workflows (or tests)
+            manager_ip = cloudify_utils.get_manager_rest_service_host()
```

Comment:
does this actually work with local workflows? that utils getter only looks in the environment, I think this is only relevant to the mgmtworker?

Reply:
Well... this is how the tests do it :) I'll see if it's relevant for local workflows

Reply:
Well, seeing how this is how it used to work so far, I feel Ok with leaving it here: https://github.com/cloudify-cosmo/cloudify-agent/blob/master/cloudify_agent/installer/config/agent_config.py#L120

Reply:
cool